### PR TITLE
Todo POST API

### PR DESCRIPTION
**Pull request by asutosh05**
## Description
Create API for todo task POST /todos

## Database schema changes
todos: {
_id: ObjectId,
title: String
}

### Automated test cases added
- POST /todos
input: {title: 'Automate test case title'}
output: success

- POST /todos
input: {title: ''}
output: error

- DELETE /todos/:id
output: success

- DELETE /todo/
output: error

- DELETE /todo/:wrongId
output: error

### Manual test cases run
- Tested using postman 
- POST /todos
Body: {
title:"Test Todo"
}
Response: { id:"some Id",title:"Test Todo"}

- DELETE
- /todos/:id
   Responce Status Code : 204
-/todos/
   Responce Status Code -404
-.todos/:wrongId
   Responce Status Code - 404

### CODE REVIEW

File: src/controllers/todo-controller.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  public basePath: string = '/todos';
+  public router: Router = Router();
+
+  constructor(ctx: AppContext) {
+    super(ctx);
+    this.initializeRoutes();
+  }
+  private initializeRoutes() {
+    this.router.post(
+            `${this.basePath}`,
```
Comment (jjalan) [inline src/controllers/todo-controller.ts]: Inconsistent spacing.

Please use 2 spaces across the board.

File: src/locales/en.json [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+		"INVALID_FIRSTNAME": "Please specify first name.",
+		"INVALID_LASTNAME": "Please specify last name.",
+		"PASSWORD_MISSING": "Please provide a password.",
+		"DUPLICATE_EMAIL": "The email you provided is already taken. Please provide a different one.",
+		"INVALID_TITLE": "VALIDATION_ERRORS.INVALID_TITLE"
+	},
+	"MESSAGES": {
+		"HEALTHCHECK": "OK"
+	}
+}
```
Comment (jjalan) [inline src/locales/en.json]: New line missing.

File: src/models/todo.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { BaseModel } from './base-model';
+import { LooseObject } from '@typings';
+
+export class Todo extends BaseModel {
```
Comment (jjalan) [inline src/models/todo.ts]: Consider renaming TODOItem or TodoItem to make it a noun rather than a verb.

File: src/models/todo.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  title: string;
+
+  constructor(json?: any) {
```
Comment (jjalan) [inline src/models/todo.ts]: 1. Why is json optional?
2. Kindly avoid any. It defeats typescript purpose. May be create a JSON or LooseObject type def.

File: test/spec/todo/todo.spec.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+before(async () => {
+    await respositoryContext.store.connect();
+    const app = new App(testAppContext);
+    app.initializeMiddlewares();
+    app.initializeControllers();
+    app.initializeErrorHandling();
+    expressApp = app.expressApp;
+})
+
+describe("POST /todos", () =>{
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts]: Space missing after `=>`

File: test/spec/todo/todo.spec.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    it("should creater a new todo when todo is not empty", async () => {
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts]: `creater` to `create`

File: config/development.json [hunk 7]
```diff
@@ -6,7 +6,7 @@
     "console": true
   },
   "mongoDb": {
-    "uri": "mongodb://localhost:27017/boilerplate-development"
+    "uri": "mongodb://127.0.0.1:27017/dev"
```
Comment (jjalan) [inline config/development.json:9]: better to keep db name as `todo-dev`. `dev` is quite generic.

**Review by jjalan — CHANGES_REQUESTED** [review]

File: src/locales/en.json [hunk 8]
```diff
@@ -1,19 +1,20 @@
...
-    "INVALID_FIRSTNAME": "Please specify first name.",
-    "INVALID_LASTNAME": "Please specify last name.",
-    "PASSWORD_MISSING": "Please provide a password.",
-    "DUPLICATE_EMAIL": "The email you provided is already taken. Please provide a different one."
-  },
-  "MESSAGES": {
-    "HEALTHCHECK": "OK"
-  }
-}
+	"DEFAULT_ERRORS": {
```
Comment (jjalan) [inline src/locales/en.json:2]: Kindly remove messages that are not being used anymore.

File: src/locales/en.json [hunk 9]
```diff
@@ -1,19 +1,20 @@
...
+		"RESOURCE_NOT_FOUND": "Requested resource not found.",
+		"VALIDATION_FAILED": "Validation failed. Please modify the request and try again."
+	},
+	"VALIDATION_ERRORS": {
+		"INVALID_EMAIL": "Please specify valid email address",
+		"INVALID_FIRSTNAME": "Please specify first name.",
+		"INVALID_LASTNAME": "Please specify last name.",
+		"PASSWORD_MISSING": "Please provide a password.",
+		"DUPLICATE_EMAIL": "The email you provided is already taken. Please provide a different one.",
+		"INVALID_TITLE": "VALIDATION_ERRORS.INVALID_TITLE"
```
Comment (balwant-jtc) [inline src/locales/en.json]: INVALID_TITLE message is not proper.

Comment (asutosh05) [reply src/locales/en.json]: proper message added

File: src/locales/en.json (hunk 8 above)
Comment (asutosh05) [reply src/locales/en.json:2]: the messages which are not in use removed.

File: config/development.json (hunk 7 above)
Comment (asutosh05) [reply config/development.json:9]: done

File: test/spec/todo/todo.spec.ts (hunk 6 above)
Comment (asutosh05) [reply test/spec/todo/todo.spec.ts]: modified

File: test/spec/todo/todo.spec.ts (hunk 5 above)
Comment (asutosh05) [reply test/spec/todo/todo.spec.ts]: space added

File: src/models/todo.ts (hunk 4 above)
Comment (asutosh05) [reply src/models/todo.ts]: now JSON is not optional

File: test/spec/todo/todo.spec.ts [hunk 10]
```diff
@@ -8,40 +8,72 @@ import chaiHttp from 'chai-http';
...
+
+  });
+  it('should return a validation error if title is empty', async () => {
+    const res = await chai.request(expressApp).post('/todos').send({
+      title: '',
+    });
+
+    expect(res).to.have.status(400);
+    expect(res.body.failures).to.have.deep.members([
+            { field: 'title', message: 'Please provide a titel.' },
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts]: `Please provide a tille`.

**Review by jjalan — APPROVED** [review]

File: test/spec/todo/todo.spec.ts (hunk 10 above)
Comment (asutosh05) [reply test/spec/todo/todo.spec.ts]: modified

File: src/models/todo.ts (hunk 3 above)
Comment (asutosh05) [reply src/models/todo.ts]: changed

File: src/locales/en.json [hunk 11]
```diff
@@ -1,19 +1,15 @@
...
-    "HEALTHCHECK": "OK"
-  }
+	"DEFAULT_ERRORS": {
+		"AUTHENTICATION_FAILED": "Please login to access this resource.",
+		"INTERNAL_SERVER_ERROR": "Internal server error. Please try again later.",
+		"RESOURCE_NOT_FOUND": "Requested resource not found.",
+		"VALIDATION_FAILED": "Validation failed. Please modify the request and try again."
+	},
+	"VALIDATION_ERRORS": {
+		"INVALID_TITLE": "Please provide a titel.",
```
Comment (balwant-jtc) [inline src/locales/en.json]: titel => title

Comment (asutosh05) [reply src/locales/en.json]: its done

**Pull request merged**